Assets/ModScripts/Submodules/WhosWho.cs

```diff
@@ -0,0 +1,293 @@
+            Module.CauseStrike();
+            return;
+        }
+        if ((Button == 2) == Info.BulbInfo[4])
+            O_PressTime.Start();
+        else
+            I_PressTime.Start();
+    }
+
+    public override void OnBulbButtonRelease(int Button)
+    {
+        O_PressTime.Stop();
+        I_PressTime.Stop();
+
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonRelease, Module.transform);
+
+        if (O_PressTime.Elapsed.TotalSeconds >= 1 || I_PressTime.Elapsed.TotalSeconds >= 1)
+        {
+            if (!submissionMode)
+            {
+                submissionMode = true;
+                UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Entering submission mode.", ModuleID);
+            }
+            else if (int.Parse(Module.WidgetText[2].text) == finalNumber)
+            {
+                UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Correctly submitted {1}. Module solved.", ModuleID, finalNumber);
+                Module.StartSolve();
+                Module.Solve();
+            }
+            else
+            {
+                UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Strike! Tried to submit {1} when the answer was {2}. Exiting submission mode.", ModuleID, Module.WidgetText[2].text, finalNumber);
+                Module.CauseStrike();
+                submissionMode = false;
+            }
+        }
+        else
+        {
+            if (!submissionMode)
+            {
+                if (Button == 3)
+                    listB_Index++;
+                else
+                {
+                    listB_Index--;
+                    listB_Index += listB.Count;
+                }
+
+                listB_Index %= listB.Count;
+                Module.WidgetText[1].text = listB[listB_Index];
+            }
+            else
+            {
+                if (Button == 3)
+                    Info.NumberDisplay++;
+                else
+                    Info.NumberDisplay += 9; // add 10 and subtract 1 so that the number is never negative
+                Info.NumberDisplay %= 10;
+                Module.WidgetText[2].text = Info.NumberDisplay.ToString();
+            }
+        }
+        O_PressTime.Reset();
+        I_PressTime.Reset();
+    }
+
+    public override void OnUtilityPress()
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.UtilityButton.GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.5f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
```

Comment:
```suggestion
        if (!Module.IsSolving && !Module.CheckValidComponents())
```
Add IsSolving so that it doesn't check every single time